addons/web/static/src/js/views/list/list_editable_renderer.js

```diff
@@ -1022,7 +1025,8 @@ ListRenderer.include({
      */
     _renderHeaderCell: function () {
         const $th = this._super.apply(this, arguments);
-        if (this.editable && this._hasVisibleRecords(this.state)) {
+        const column = this.columns.find(col => col.attrs.name && col.attrs.name === $th.data('name'));
```

Comment:
I think that you receive the column in argument (altough it's named 'node' in the ListRenderer). So for instance you could rather check that `node.tagName === 'field'`

Reply:
After giving it some thoughts I think it is best to simply ignore empty headers (no inner html), it makes less sense to resize those columns